Dockerfile

```diff
@@ -1,66 +1,134 @@
+RUN RUSTFLAGS="-Ctarget-feature=-crt-static" cargo chef cook --release -p lldap
+RUN RUSTFLAGS="-Ctarget-feature=-crt-static" cargo chef cook --release -p migration-tool
+COPY . /lldap-src
+WORKDIR /lldap-src
 
-COPY --from=builder /app/app/index_local.html app/index.html
-COPY --from=builder /app/app/static app/static
-COPY --from=builder /app/app/pkg app/pkg
-COPY --from=builder /app/target/release/lldap /app/target/release/migration-tool ./
-COPY docker-entrypoint.sh lldap_config.docker_template.toml ./
+# Compiling application, take your time
+### amd64
+RUN cargo build --target=x86_64-unknown-linux-gnu --release -p lldap -p migration-tool
+#######################################################
+### Only enable if building non-native architecture ###
+#######################################################
+### arm64
+#RUN cargo build --target=aarch64-unknown-linux-gnu --release -p lldap -p migration-tool
+### armhf
+#RUN cargo build --target=armv7-unknown-linux-gnueabihf --release -p lldap -p migration-tool
 
+
+### Build frontend
+RUN app/build.sh
+# Prepare our application path
+RUN mkdir -p /lldap/app
+# Web and App dir
+COPY docker-entrypoint.sh /docker-entrypoint.sh
+COPY lldap_config.docker_template.toml /lldap/
+# The applications
+RUN cp target/*/release/lldap /lldap/lldap && \
+    cp target/*/release/migration-tool /lldap/migration-tool && \
+    cp -R app/index.html \
+          app/pkg \
+          app/static \
+          /lldap/app/
+# Just checking
+RUN ls -al /lldap && \
+    ls -al /lldap/app
+# Fetch our fonts
+WORKDIR /lldap
 RUN set -x \
-    && apk add --no-cache bash \
-    && for file in $(cat app/static/libraries.txt); do wget -P app/static "$file"; done \
-    && for file in $(cat app/static/fonts/fonts.txt); do wget -P app/static/fonts "$file"; done \
+    && for file in $(cat /lldap/app/static/libraries.txt); do wget -P app/static "$file"; done \
+    && for file in $(cat /lldap/app/static/fonts/fonts.txt); do wget -P app/static/fonts "$file"; done \
     && chmod a+r -R .
 
-ENV LDAP_PORT=3890
-ENV HTTP_PORT=17170
 
-EXPOSE ${LDAP_PORT} ${HTTP_PORT}
 
-ENTRYPOINT ["/app/docker-entrypoint.sh"]
+### Final image
+FROM alpine:3.16
+WORKDIR /app
+ENV UID=1000
+ENV GID=1000
+ENV USER=lldap
+RUN echo "http://dl-cdn.alpinelinux.org/alpine/edge/testing/" >> /etc/apk/repositories && \
+    apk add --no-cache tini ca-certificates bash gosu && \
+    addgroup -g $GID $USER && \
+    adduser \
+    --disabled-password \
+    --gecos "" \
+    --home "$(pwd)" \
+    --ingroup "$USER" \
+    --no-create-home \
+    --uid "$UID" \
+    "$USER" && \
+    mkdir -p /data && \
+    chown $USER:$USER /data
+### Copy out the binary and web from builder
+COPY --from=builder --chown=$USER:$USER /lldap /app
+COPY --from=builder --chown=$USER:$USER /docker-entrypoint.sh /docker-entrypoint.sh
+VOLUME ["/data"]
+WORKDIR /app
+ENTRYPOINT ["tini", "--", "/docker-entrypoint.sh"]
```

Comment:
I think we don't need both the entrypoint and the CMD, you can merge them into an entrypoint.